apps/builder/app/shared/pubsub/create.ts

```diff
@@ -27,32 +27,36 @@ export const createPubsub = <PublishMap>() => {
     );
   }
 
+  const iframes = new Set<HTMLIFrameElement>();
+  const registerIframe = (element: HTMLIFrameElement) => {
```

Comment:
would it make sense to not couple iframe and pubsub and instead just register "an object" that has a postMessage method or even a function that returns that object

would make sense if we already know we want to be able to subscribe also without using an iframe

Reply:
or even registering directly a postMessage function that would be called

Reply:
No guarantee postMessage is bound and can be called without "this".

Reply:
yeah, can be a callback registerPostMessage((...args) => {
  iframe.contentWindow.postMessage(...args)
})

Reply:
would call that something else than registerPostMessage though

Reply:
could be just .addEventListener or .on((...args) => {
   iframe.contentWindow.postMessage(...args)
})